sympy/matrices/matrices.py

```diff
@@ -1053,9 +1053,26 @@ def extract(self, rowsList, colsList):
         [2],
         [8]])
 
+        when colsList or rowsList is boolean then indexing will start at 0.
```

Comment:
You should put a description of what it does when there are boolean values here. Something like "rowsList or colsList can also be a list of booleans, in which case the rows or columns corresponding to the True values will be selected".